packages/selected-network-controller/tests/SelectedNetworkController.test.ts

```diff
@@ -558,6 +559,53 @@ describe('SelectedNetworkController', () => {
         });
       });
     });
+    describe('when the domain is a snap (starts with "npm:" or "local:")', () => {
+      it('returns a proxied globally selected networkClient and does not create a new proxy in the domainProxyMap', () => {
+        const { controller, domainProxyMap, messenger } = setup({
+          state: {
+            domains: {},
+          },
+          useRequestQueuePreference: true,
+        });
+        jest.spyOn(messenger, 'call');
+        const snapDomain = 'npm:@metamask/bip32-example-snap';
+        const result = controller.getProviderAndBlockTracker(snapDomain);
+        expect(domainProxyMap.get(snapDomain)).toBeUndefined();
+        expect(messenger.call).toHaveBeenCalledWith(
```

Comment:
Nit: Hmm, this feels like testing internal implementation details, rather than testing the behavior we want. In theory we might refactor this to use a new equivalent action, or we might end up calling this in other code paths, or we might rename this action, etc.

To make this test less fragile, we could try switching the globally selected network and test that the provider we got has the correct chain before and after the switch. Maybe that would work, that's just the first suggestion that comes to mind.

Reply:
I totally agree that this is not a great way to test it. Testing it the right way involves pulling in the actual `NetworkController` which I was hoping to do at a later time 😅 . I put this TODO in there about this same point a few weeks ago: https://github.com/MetaMask/core/blob/9687bfcbb03442276ad1e7325b358e087ccbb4e0/packages/selected-network-controller/tests/SelectedNetworkController.test.ts#L697

Reply:
added the TODO in another spot here: https://github.com/MetaMask/core/pull/4259/commits/0fd8b54cbdf7e372706317f285ce3495eea3ad16

Reply:
Hmm, we've been trying to move away from using actual full controllers as unit test fixtures as that tends to be fragile as well for different reasons (any breaking change to controller construction, initialization, etc. breaks unrelated tests), and sometimes they have internals that aren't well suited to unit tests that are difficult to mock out (e.g. network requests).

Leaving this improvement as a TODO sounds good to me though!